### PR TITLE
First iteration of summary rollup.

### PR DESCRIPTION
**Pull request by kevinliang43**
### Daily, Weekly, Monthly Aggregation
Example Raw Data
image

Example Aggregated Data:
image

## Method Of Aggregation
How this works:

1. Pull in data from `raw_item_ts`, filtered by `item_id`, `start_ts`, `end_ts` (calculated by `start_ts`+period) and ordering by `start_ts` ASC (earlier timestamps come first)

2. Iterate through the rows:
  - For each pair of rows (`row1`, `row2`):
    - if `row1.state` == `row2.state`, then we increment the `state`'s total_time by (`row1.end_ts` - `row2.end_ts`) 
    - else increment `row2.state`'s time by (`row2.end_ts` - `row2.start_ts`)

3. Write

## Comments
<s>First Iteration. Still need to add support for `Month` in our aggregations. As of now, we can support periods of 'day', and 'week' for our aggregations. </s>

(11/14/19) - Functionality for `day`, `week`, and `month` aggregations are currently supported.

Things I want to add:
1. More logging
2. More comments
3. Clean up the code.

However, this will be good as a first iteration for our next demo.

### CODE REVIEW

**Comment by kevinliang43** [comment]
Issue with Bill.sql schema for `raw_item_ts`: https://github.com/BU-NU-CLOUD-F19/MOC_Reporting/issues/22

Holding off from merging/reviewing for now.

**Comment by tsachleben** [comment]
Issue #22 fixed by #23

**Comment by kevinliang43** [comment]
Added functionality for monthly aggregations. Refactored aggregations to be on `item_id` rather than `catalog_item_id`. 

Merging of this PR is dependent on this PR being merged first: https://github.com/BU-NU-CLOUD-F19/MOC_Reporting/pull/25

**Review by tsachleben — COMMENTED** [review]
Will want to discuss in-person

File: summary_rollup.py
Comment (tsachleben) [inline summary_rollup.py]: Add edge condtion:

> if raw_item_ts_rows is not None and len(raw_item_ts_rows) > 0:

**Review by anju-james — DISMISSED** [review]
Good to merge once the comments above are addressed.

File: summary_rollup.py
Comment (tsachleben) [inline summary_rollup.py]: Need to be able to signal to top level when a write has failed.

**Review by tsachleben — CHANGES_REQUESTED** [review]
See comments in code

**Comment by kevinliang43** [comment]
Added edge case check for querying raw_item_ts rows.

Added signaling to top level when reads/writes fail.

**Review by anju-james — APPROVED** [review]
LGTM

**Review by tsachleben — APPROVED** [review]

**Pull request merged**
